src/dummy.rs

```diff
@@ -19,6 +19,131 @@ use erg_compiler::Compiler;
 pub type EvalError = CompileError;
 pub type EvalErrors = CompileErrors;
 
+/// The instructions for communication between the client and the server.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[repr(u8)]
+enum Inst {
+    /// Send from server to client. Informs the client to print data.
+    Print = 0x01,
+    /// Send from client to server. Informs the REPL server that the executable .pyc file has been written out and is ready for evaluation.
+    Load = 0x02,
+    /// Send from server to client. Represents an exception.
+    Exception = 0x03,
+    /// Send from server to client. Tells the code generator to initialize due to an error.
+    Initialize = 0x04,
+    /// Informs that the connection is to be / should be terminated.
+    Exit = 0x05,
+    /// Informs that it is not a supported instruction.
+    Unknown = 0x00,
+}
+
+impl From<u8> for Inst {
+    fn from(v: u8) -> Inst {
+        match v {
+            0x01 => Inst::Print,
+            0x02 => Inst::Load,
+            0x03 => Inst::Exception,
+            0x04 => Inst::Initialize,
+            0x05 => Inst::Exit,
+            _ => Inst::Unknown,
+        }
+    }
+}
+
+/// -------------------------------
+/// | ins    | size    | data
+/// -------------------------------
+/// | 1 byte | 2 bytes | n bytes
+/// -------------------------------
+#[derive(Debug, Clone)]
+struct Message {
+    inst: Inst,
+    size: usize,
```

Comment:
`size` should be `u16`.

Reply:
`u16::MAX == 65535`. I think it is sufficient, but if it seems to be short, we should change it to `u32` or something.

Reply:
> `size` should be `u16`.

Using `usize` here is because in Rust, the `usize` type is typically used to represent the length of data.

> `u16::MAX == 65535`. I think it is sufficient, but if it seems to be short, we should change it to `u32` or something.

`u16` for encoding is sufficient for a REPL in Rust.

Reply:
```
-------------------------------
| ins    | size    | data 
-------------------------------
| 1 byte | 2 bytes | n bytes
-------------------------------
```

We defined the size of the length field to be 2 bytes. `usize` is 4~8 bytes, and the length varies from platform to platform.

Reply:
I think that the type used for low-level data transmission is independent of the object representation used in the application level. For example, you can use utf8/utf16 to transmit a string, but use String to represent it in the application level.

If you are concerned about unsafe type conversions, I suggest checking the data length during encoding to ensure it does not exceed the u16 max.

Reply:
No, `usize.to_be_bytes()` and `u16.to_be_bytes()` return different byte arrays.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1e11525e1ac0bb003547ade0b5d9bf16

Reply:
If the data exceeds the `u16` range, the data can be divided into multiple packets and sent.

Reply:
> No, `usize.to_be_bytes()` and `u16.to_be_bytes()` return different byte arrays.
> 
> [play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1e11525e1ac0bb003547ade0b5d9bf16](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1e11525e1ac0bb003547ade0b5d9bf16)

 :ok_hand: I will change it.



> If the data exceeds the `u16` range, the data can be divided into multiple packets and sent.

We are currently using 2 bytes to represent the length, and if we want to support larger data, we need to change the protocol format.

Reply:
Fixed.